app/templates/main/map.html

```diff
@@ -106,509 +111,630 @@
+          //   errString += "No file has been uploaded. Please select a file.";
+          // }
+          if (Object.keys(boxes).length <= 0) {
+            errString += " Please select at least one bounding box."
+          }
+          if (!$('#filename').val()) {
+            errString += " Please enter a name for your file."
+          }
+          if ($("#startDay").val() === "") {
+            errString += " Please select a valid start day"
+          }
+          if ($("#endDay").val() === "") {
+            errString += " Please select a valid end day."
+          }
+
+          if (errString.length > 0) {
+            $('#submitProcess').before("<div id='submitErrors' class='col-md-12'><div class='alert alert-danger'>" + errString + "<\/div><\/div>");
+          } else {
+            obj = processData(uploadURL, boxes, $('#email').val(), $('#startDay').val(), $('#endDay').val());
+            inside = new Array(obj.boxes.length).fill(false);
+            startTimes = new Array(obj.boxes.length).fill(null);
+            durations = new Array(obj.boxes.length).fill(0);
+            var newOboe = oboe();
+
+            newOboe.node('locations.*', function (location) {
+              if (weekEndMS === 0) {
+                weekEndMS = moment.unix(location.timestampMs / 1000).day(obj.dayEnd).hour(23).minute(59).unix() * 1000;
+                weekStartMS = moment.unix(location.timestampMs / 1000).day(obj.dayStart).hour(0).minute(0).unix() * 1000;
+              }
+
+              if (location.timestampMs >= weekStartMS && location.timestampMs <= weekEndMS) {
+                // within a week
+                for (var i = 0; i < obj.boxes.length; i++) {
+                  // iterate thru the boxes
+                  var box = obj.boxes[i];
+                  var lat = location.latitudeE7 / Math.pow(10, 7)
+                  var lng = location.longitudeE7 / Math.pow(10, 7)
+                    // check if in box coors && not inside a box
+                  if (box['swLat'] <= lat && box['neLat'] >= lat &&
+                    box['swLng'] <= lng && box['neLng'] >= lng && !inside[i]) {
+                    console.log('enetered ' + box.name);
+                    inside[i] = true;
+                    startTimes[i] = location.timestampMs;
+                  } else if ((box['swLat'] <= lat && box['neLat'] >= lat &&
+                      box['swLng'] <= lng && box['neLng'] >= lng) && !inside[i]) {
+                    durations[i] = Math.abs(location.timestampMs - startTimes[i]);
+                  }
+                  // not inside the box
+                  else if (!(box['swLat'] <= lat && box['neLat'] >= lat &&
+                      box['swLng'] <= lng && box['neLng'] >= lng) && inside[i]) {
+                    inside[i] = false;
+                    durations[i] = Math.abs(location.timestampMs - startTimes[i]);
+                  }
+                }
+
+              }
+              else if (location.timestampMs < weekStartMS) {
```

Comment:
i think putting a comment here saying that this is the end of a work week would be helpful